app/src/main/java/com/halim/adam/cashmaster/ViewBudgets.java

```diff
@@ -24,43 +24,51 @@ protected void onCreate(Bundle savedInstanceState) {
 
         // get data
         ArrayList<BudgetRatio> budgetRatioList = dbHelper.GetBudgetRatioList();
-        final ArrayList<Integer> budgetIdList = new ArrayList<>();
-        ArrayList<String> budgetNameList = new ArrayList<>();
-        ArrayList<Float> ratioList = new ArrayList<>();
-        ArrayList<Float> budgetTotalList = new ArrayList<>();
-
-        for(int c = 0; c < budgetRatioList.size(); c++){
-            budgetIdList.add(budgetRatioList.get(c).getId());
-            budgetNameList.add(budgetRatioList.get(c).getName());
-            ratioList.add(budgetRatioList.get(c).getRatio());
-
-            // get budget total
-            ArrayList<Budget> budgetList = dbHelper.GetBudgetFromRatio(budgetRatioList.get(c).getId());
-            float budgetTotal = 0;
-            if(budgetList != null){
-                budgetTotal += budgetList.get(c).getAmount();
+        Integer[] idArray;
+        String[] nameArray;
+        Float[] ratioArray;
+        Float[] totalArray;
+
+        if(budgetRatioList != null){
+            idArray = new Integer[budgetRatioList.size()];
+            nameArray = new String[budgetRatioList.size()];
+            ratioArray = new Float[budgetRatioList.size()];
+            totalArray = new Float[budgetRatioList.size()];
+
+            for(int c = 0; c < budgetRatioList.size(); c++){
+                idArray[c] = budgetRatioList.get(c).getId();
+                nameArray[c] = budgetRatioList.get(c).getName();
+                ratioArray[c] = budgetRatioList.get(c).getRatio();
+
+                // get budget total
+                ArrayList<Budget> budgetList = dbHelper.GetBudgetFromRatio(budgetRatioList.get(c).getId());
+                float budgetTotal = 0;
```

Comment:
ArrayList.get(int index) is O(1), but if you do it 4 times in a loop, you are doing extra operations while you can actually do it only once and save the object in a variable and just use the getter methods, because you are calling the get() method for the same object (index of c). This will also make the code shorter and easier to read.